docs/examples/webextension/background.js

```diff
@@ -0,0 +1,14 @@
+fetch('https://testpilot.firefox.com/api/metrics/ping/testpilottest', {
+  method: 'POST',
+  mode: 'cors',
+  headers: {'Content-Type': 'application/json'},
+  body: JSON.stringify({
+    "some": "random",
+    "metrics": "ping",
+    "payload": "here"
+  })
+}).catch(e => {
+  console.log('problem sending metrics ping', e);
+}).then(resp => {
+  console.log('metrics ping success', resp);
+});
```

Comment:
Turns out the `catch()` in the middle lets you handle the error and continue, so it'd have done both `.catch()` and `.then()`:

```js
function promiseTest(data = new Error("uh oh!")) {
  return new Promise((resolve, reject) => {
    if (data instanceof Error) {
      throw data;
    }
    resolve(data);
  });
}

promiseTest( /* error */ )
  .then((data) => {
    console.log(`first then(): ${data}`);
    return data;
  }).catch((err) => {
    console.error(`An error? ${err}`);
    return err;
  }).then((data) => {
    console.info(`after catch()? ${data}`);
  });
```

```sh
$ node promise.test.js

An error? Error: uh oh!
after catch()? Error: uh oh!
```